modules/linear_boltzmann_solvers/executors/pi_keigen_smm.cc

```diff
@@ -0,0 +1,1122 @@
+            {
+              const auto pwld_dof = pwld_global_to_local_map.at(pwld_global_dof);
+              const auto pwlc_dof = pwlc_global_to_local_map.at(pwlc_global_dof);
+
+              pwlc_wtd_avg[pwlc_dof] += ghosted_pwld_vector[pwld_dof] * vol;
+              pwlc_vol_sum[pwlc_dof] += vol;
+            }
+          }
+  }
+
+  // Compute nodal averages
+  for (int k = 0; k < num_local_pwlc_dofs; ++k)
+    pwlc_wtd_avg[k] /= pwlc_vol_sum[k];
+
+  // Project the PWLC vector back to PWLD
+  std::vector<double> output = pwld_vector;
+  for (const auto& cell : grid.local_cells)
+  {
+    const auto& cell_mapping = pwld.GetCellMapping(cell);
+    for (int i = 0; i < cell_mapping.NumNodes(); ++i)
+      for (int u = 0; u < uk_man.unknowns_.size(); ++u)
+        for (int c = 0; c < uk_man.unknowns_[u].num_components_; ++c)
+        {
+          const auto pwld_dof = pwld.MapDOFLocal(cell, i, uk_man, u, c);
+          const auto pwlc_dof = pwlc.MapDOFLocal(cell, i, uk_man, u, c);
+          output[pwld_dof] = pwlc_wtd_avg[pwlc_dof];
+        }
+  }
+  return output;
+}
+
+std::vector<int64_t>
+PowerIterationKEigenSMM::MakePWLDGhostIndices(const SpatialDiscretization& pwld,
+                                              const UnknownManager& uk_man)
+{
+  std::set<int64_t> ghost_ids;
+  const auto& grid = pwld.Grid();
+  for (const uint64_t ghost_id : grid.cells.GetGhostGlobalIDs())
+  {
+    const auto& cell = grid.cells[ghost_id];
+    const auto& cell_mapping = pwld.GetCellMapping(cell);
+    for (int i = 0; i < cell_mapping.NumNodes(); ++i)
+      for (int u = 0; u < uk_man.NumberOfUnknowns(); ++u)
+        for (int c = 0; c < uk_man.unknowns_[u].NumComponents(); ++c)
+          ghost_ids.insert(pwld.MapDOF(cell, i, uk_man, u, c));
+  }
+  return std::vector<int64_t>(ghost_ids.begin(), ghost_ids.end());
+}
+
+PowerIterationKEigenSMM::GhostInfo
+PowerIterationKEigenSMM::MakePWLDGhostInfo(const SpatialDiscretization& pwld,
+                                           const UnknownManager& uk_man)
+
+{
+  const auto num_local_dofs = pwld.GetNumLocalDOFs(uk_man);
+  const auto num_global_dofs = pwld.GetNumGlobalDOFs(uk_man);
+
+  // Build list of global IDs
+  auto ghost_ids = MakePWLDGhostIndices(pwld, uk_man);
+
+  // Create the ghost communicator
+  auto vec_ghost_comm = std::make_shared<VectorGhostCommunicator>(
+    num_local_dofs, num_global_dofs, ghost_ids, opensn::mpi_comm);
+
+  // Create the mapping
+  int64_t k = 0;
+  std::map<int64_t, int64_t> ghost_global_to_local_map;
+  for (const int64_t ghost_id : ghost_ids)
+    ghost_global_to_local_map[ghost_id] = static_cast<int64_t>(num_local_dofs + k++);
+
+  return {vec_ghost_comm, ghost_global_to_local_map};
+}
+
+int
+PowerIterationKEigenSMM::MapAssociatedFaceNode(const Vector3& node,
+                                               const std::vector<Vector3>& nbr_nodes,
+                                               const double epsilon /* = 1.0e-12 */)
```

Comment:
Remove `/* = 1.0e-12 */`, it is in the header file. 

The reason why one does not want to do this is that sometime in future, somebody is gonna change the value in the header file and they won't change it here and it would create inconsistency.

Reply:
this is an example of a comment that, I agree, can be removed